datadog_checks_dev/datadog_checks/dev/tooling/manifest_validator/v1/validator.py

```diff
@@ -0,0 +1,156 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import os
+import uuid
+
+import datadog_checks.dev.tooling.manifest_validator.common.validator as common
+from datadog_checks.dev.tooling.constants import get_root
+from datadog_checks.dev.tooling.manifest_validator.common.validator import BaseManifestValidator
+from datadog_checks.dev.tooling.manifest_validator.v1.schema import get_manifest_schema
+from datadog_checks.dev.tooling.utils import has_logs, is_package, parse_version_parts
+
+
+class AttributesValidator(BaseManifestValidator):
+    """Check that attributes are valid"""
+
+    def validate(self, check_name, decoded, fix):
+        errors = sorted(get_manifest_schema().iter_errors(decoded), key=lambda e: e.path)
+        if errors:
+            for error in errors:
+                self.fail(f'  {"->".join(map(str, error.absolute_path))} Error: {error.message}')
+
+
+class GUIDValidator(BaseManifestValidator):
+    all_guids = {}
+
+    def validate(self, check_name, decoded, fix):
+        guid = decoded.get('guid')
+        if guid in self.all_guids:
+            output = f'  duplicate `guid`: `{guid}` from `{self.all_guids[guid]}`'
+            if fix:
+                new_guid = uuid.uuid4()
+                self.all_guids[new_guid] = check_name
+                decoded['guid'] = new_guid
+                self.fix(output, f'  new `guid`: {new_guid}')
+            else:
+                self.fail(output)
+        elif not guid or not isinstance(guid, str):
+            output = '  required non-null string: guid'
+            if fix:
+                new_guid = uuid.uuid4()
+                self.all_guids[new_guid] = check_name
+                decoded['guid'] = new_guid
+                self.fix(output, f'  new `guid`: {new_guid}')
+            else:
+                self.fail(output)
+        else:
+            self.all_guids[guid] = check_name
+        return self.result
+
+
+class ManifestVersionValidator(BaseManifestValidator):
+    def __init__(self, *args, **kwargs):
+        super(ManifestVersionValidator, self).__init__(*args, **kwargs)
+        self.root = get_root()
+
+    def validate(self, check_name, decoded, fix):
```

Comment:
I'm a bit confused on this validator, is this tying the manifest version to the integrations version defined in about.py?

Reply:
Yeah, that's why I left it with the V1 validations, not sure how helpful it will be for V2 and beyond.